internal/fetchers/oci/manifest/manifest.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The Enterprise Contract Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package manifest
+
+import (
+	"archive/tar"
+	"context"
+	"encoding/json"
+	"io"
+	"path"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+
+	"github.com/enterprise-contract/ec-cli/internal/fetchers/oci"
+)
+
+func ImageManifests(ctx context.Context, ref name.Reference, opts ...remote.Option) (map[string]json.RawMessage, error) {
+	img, err := oci.NewClient(ctx).Image(ref, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	content := mutate.Extract(img)
+	defer content.Close()
+	archive := tar.NewReader(content)
+
+	manifests := map[string]json.RawMessage{}
+	for {
+		header, err := archive.Next()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, err
+		}
+
+		name := header.Name
+
+		// we're only interested in files in manifests/*
+		if !strings.EqualFold(path.Dir(name), "manifests") {
+			continue
+		}
+
+		// we're only interested in JSON or YAML files
+		if ext := path.Ext(name); !strings.EqualFold(ext, ".yaml") && !strings.EqualFold(ext, ".json") {
+			continue
+		}
+
+		// TODO: large files could be an issue. We do need to read the archive
```

Comment:
[io.LimitedReader](https://pkg.go.dev/io#LimitedReader) might be something to consider later on.

Reply:
I much rather have a (OPA) function and let the policy pull in what it needs, if we had a metric of how big the manifest files can get then perhaps, left a TODO to think about this in the future...

Reply:
works for now.